internal/metamorphic/crossversion/crossversion_test: TestMetaCrossVersion failed

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/11516766?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/11516766?buildTab=artifacts#/) on refs/heads/master @ [0b401ee526b8](https://github.com/cockroachdb/pebble/commits/0b401ee526b8):


```
                --- PASS: TestMeta/execution/random-020 (6.71s)
                --- PASS: TestMeta/execution/random-023 (8.50s)
                --- PASS: TestMeta/execution/random-019 (3.82s)
                --- PASS: TestMeta/execution/random-017 (5.79s)
                --- PASS: TestMeta/execution/random-013 (5.01s)
                --- PASS: TestMeta/execution/random-012 (6.07s)
                --- PASS: TestMeta/execution/random-014 (8.29s)
                --- PASS: TestMeta/execution/random-018 (11.38s)
                --- PASS: TestMeta/execution/random-011 (5.87s)
                --- PASS: TestMeta/execution/random-009 (5.73s)
                --- PASS: TestMeta/execution/random-010 (7.61s)
                --- PASS: TestMeta/execution/random-006 (5.14s)
                --- PASS: TestMeta/execution/random-021 (22.45s)
                --- PASS: TestMeta/execution/random-015 (16.01s)
                --- PASS: TestMeta/execution/random-008 (7.63s)
                --- PASS: TestMeta/execution/random-007 (6.73s)
                --- PASS: TestMeta/execution/standard-013 (5.31s)
                --- PASS: TestMeta/execution/standard-025 (5.44s)
                --- PASS: TestMeta/execution/random-005 (9.62s)
                --- PASS: TestMeta/execution/standard-023 (3.98s)
                --- PASS: TestMeta/execution/standard-024 (5.94s)
                --- PASS: TestMeta/execution/standard-022 (5.53s)
                --- PASS: TestMeta/execution/standard-019 (2.14s)
                --- PASS: TestMeta/execution/random-003 (15.70s)
                --- PASS: TestMeta/execution/standard-020 (4.46s)
                --- PASS: TestMeta/execution/random-001 (18.45s)
                --- PASS: TestMeta/execution/random-002 (19.23s)
                --- PASS: TestMeta/execution/standard-018 (4.25s)
                --- PASS: TestMeta/execution/random-004 (21.28s)
                --- PASS: TestMeta/execution/standard-016 (4.85s)
                --- PASS: TestMeta/execution/standard-017 (5.41s)
                --- FAIL: TestMeta/execution/standard-011 (0.36s)
                --- PASS: TestMeta/execution/standard-014 (4.84s)
                --- PASS: TestMeta/execution/standard-007 (4.75s)
                --- PASS: TestMeta/execution/standard-012 (4.96s)
                --- PASS: TestMeta/execution/standard-009 (4.64s)
                --- PASS: TestMeta/execution/standard-010 (5.27s)
                --- PASS: TestMeta/execution/standard-021 (16.49s)
                --- PASS: TestMeta/execution/random-016 (43.80s)
                --- PASS: TestMeta/execution/standard-004 (4.49s)
                --- PASS: TestMeta/execution/standard-006 (3.89s)
                --- PASS: TestMeta/execution/standard-008 (5.06s)
                --- PASS: TestMeta/execution/standard-015 (11.04s)
                --- PASS: TestMeta/execution/standard-005 (3.86s)
                --- PASS: TestMeta/execution/standard-002 (3.38s)
                --- PASS: TestMeta/execution/standard-001 (3.09s)
                --- PASS: TestMeta/execution/standard-003 (3.84s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.1/sha=2c3ac2dc-seed=692538100799834142-opts=random-018(sha=d1171c6f-seed=5957816622292119842-opts=random-018(sha=4ddacdaf-seed=15246946898130383850-opts=random-011(<empty>))) (446.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/sstable/reader.go
+++ b/sstable/reader.go
@@ -1973,7 +1973,7 @@ func (i *twoLevelIterator) SeekPrefixGE(
 	// block load.
 
 	var dontSeekWithinSingleLevelIter bool
-	if i.topLevelIndex.isDataInvalidated() || !i.topLevelIndex.valid() || !i.index.isDataInvalidated() || err != nil ||
+	if i.topLevelIndex.isDataInvalidated() || !i.topLevelIndex.valid() || i.index.isDataInvalidated() || err != nil ||
 		(i.boundsCmp <= 0 && !flags.TrySeekUsingNext()) || i.cmp(key, i.topLevelIndex.Key().UserKey) > 0 {
 		// Slow-path: need to position the topLevelIndex.
 

